service/qos/qos_common.hh

```diff
@@ -30,6 +30,8 @@ namespace qos {
 
 enum class include_effective_names { yes, no };
 
+enum class query_context { group0, user, unspecified };
```

Comment:
Some comment explaining what the different context are would be helpful. Especially, it isn't clear what is the `unspecified` context (that the distinguish is not important).

Reply:
I added a comment to explain the enum